<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="UTF-8">
    <meta http-equiv="X-UA-Compatible" content="IE=edge">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <title>try catch</title>
</head>
<body>
    <script>
        function fn1(){
            console.log('fn1의 시작');
            try{
                fn2();
            }
            catch(e){
                console(e)
            }
            console.log('fn1의 끝');
        }
        function fn2(){
            console.log('fn2의 시작');
            console.log('fn2의 끝');
            // throw '고의적 에러';    //여기서부터 프로그램끝까지 에러 - try catch사용해야함
            throw new Error('고의로 낸 에러');
        }
        console.log('===프로그램 시작===');
        fn1();
        console.log('===프로그램 끝===');
    </script>
</body>
</html>